README.md

```diff
@@ -1,16 +1,122 @@
\ No newline at end of file
+## ⚠️ Before you start 
+
+Before you start using the template, you probably want to install the [Jetpack Compose UI Architecture Plugin](https://plugins.jetbrains.com/plugin/19034-jetpack-compose-ui-architecture-templates) on Android Studio.
+
+This plugins facilitates the work when creating new Compose screens.
+
+After installing, creating new features in the correct template structure should be as easy as doing the following:
+
+**1 - Select the "New feature" option**
+
+![](https://user-images.githubusercontent.com/8679058/223730854-7e199e33-1b99-49a3-ae1b-30a6ea725681.png)
+
+
+**2 - Name your new feature**
+
+![](https://user-images.githubusercontent.com/8679058/223731483-d0fce3f9-dba4-459d-9bb5-65e76abe2ee6.png)
+
+
+**3 - The correct folder structure should be created for you**
+
+![](https://user-images.githubusercontent.com/8679058/223731540-1604198c-78ac-4b3b-ad84-b3fc290746af.png)
+
+
+
+## Project Structure
+Android template project has three top-level packages: `core`, `designsystem` and `features`. Overall package structure looks like following
+- `core`
+  - `extensions`
+  - `injection`
+  - `network`
+  - `persistence`
+  - `ui`
+- `designsystem`
+  - `components`
+  - `theme`
+- `features`
+  - `nstack`
+  - `feature1`
+    - `data`
+    - `domain`  
+    - `injection`
+    - `ui`
+  
+### `core` package
+Core package is meant to be Domain-agnostic. It means that it should never reference anything Project specific. Instead, it serves as a home for components that can be part of any project. This includes extensions on Kotlin Classes, Utility classes, Base classes, Compose layouts and Compose Effects and Modifiers that are Behaviroual rather then UI emmiting
+
+### `designsystem` package
+The design system package contains the building blocks  for your application's UI. The main thing you will find here is `Theme`. This template is using custom `Theme` provider for Jetpack Compose while having `Material Theme` as its foundation. This allows to tweak `Colors` and `Typography` so it reflects actual Design System that is used on the project and supply custom properties like `Dimensions`
+
+Another part of the design system are components. Examples of `designsystem` components are Buttons, TopBars, TextFields and so on. Template provides `AppButton` `AppTextField` and `AppTopBar` that you can modify and tweak and use throughout you project. [Learn more about design system here](./docs/DESIGN_SYSTEM.md)
+
+
+### `features` package
+Features package contains the most important part of any projec. A Feature typically contains all the necessary code, resources, and assets required to implement a specific application functionality, such as a login screen, shopping cart, or authentication.  It may also have its own sets of dependencies, like third-party libraries or other features. [Each feature is following Clean Architecture Principles and has a three-layer split](#architecture)
```

Comment:
```suggestion
Features package contains the most important part of any project. A Feature typically contains all the necessary code(and resources + assets if migrated to multi module) required to implement a specific application functionality, such as a login screen, shopping cart, or authentication.  It may also have its own sets of dependencies, like third-party libraries or other features. [Each feature is following Clean Architecture Principles and has a three-layer split](#architecture)
```